solutions/fscloud/variables.tf

```diff
@@ -0,0 +1,60 @@
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API Key"
+  sensitive   = true
+}
+
+variable "region" {
+  type        = string
+  description = "Region to provision all resources created by this DA"
+  default     = "us-south"
+}
+
+variable "prefix" {
+  type        = string
+  description = "Prefix to append to all resources created by this DA"
+}
+
+variable "resource_group" {
+  type        = string
+  description = "An existing resource group name to use for this DA, if unset a new resource group will be created"
+  default     = null
+}
+
+variable "resource_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to created resources"
+  default     = []
+}
+
+variable "kms_key_crn" {
+  type        = string
+  description = "The root key CRN of a Key Management Services like Key Protect or Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption."
+}
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Hyper Protect or Key Protect instance in which the key specified in `kms_key_crn` is coming from."
+  type        = string
+}
+
+variable "users" {
+  description = "List of users to add."
+  type        = list(string)
+  default     = []
+}
+
+variable "resource_keys" {
```

Comment:
We need to figure out how to handle this for a DA. Exposing an object variable in a DA is not ideal - input variables should be more simple for DAs. Please check with the team how we want to handle this. One suggestion was exposing 3 boolean variables `create_writer_resource_key`, `create_reader_resource_key`, `create_manager_resource_key` and have outputs for each. But whatever approach we go with, we should use it for all DAs. The same question is open for COS DA

Reply:
As per the deep dive discussion, keeping it as is for now